pkg/apis/core/validation/validation.go

```diff
@@ -3395,15 +3399,18 @@ func validateResizePolicy(policyList []core.ContainerResizePolicy, fldPath *fiel
 			allErrors = append(allErrors, field.NotSupported(fldPath, p.ResourceName, sets.List(supportedResizeResources)))
 		}
 		switch p.RestartPolicy {
-		case core.NotRequired, core.RestartContainer:
+		case core.ResizeRestartPolicyPreferNoRestart,
+			core.ResizeRestartPolicyNotRequired,
+			core.ResizeRestartPolicyRestartContainer:
 		case "":
 			allErrors = append(allErrors, field.Required(fldPath, ""))
 		default:
 			allErrors = append(allErrors, field.NotSupported(fldPath, p.RestartPolicy, sets.List(supportedResizePolicies)))
 		}
 
-		if *podRestartPolicy == core.RestartPolicyNever && p.RestartPolicy != core.NotRequired {
-			allErrors = append(allErrors, field.Invalid(fldPath, p.RestartPolicy, "must be 'NotRequired' when `restartPolicy` is 'Never'"))
+		if *podRestartPolicy == core.RestartPolicyNever && p.RestartPolicy == core.ResizeRestartPolicyRestartContainer {
```

Comment:
You still need to handle the deprecated value, right?

Reply:
No. The logic was sort of backwards before, saying the restart policy "must be 'NotRequired' when 'restartPolicy' is 'Never'". It's more accurate to say it "cannot be RestartContainer when 'restartPolicy' is 'Never'", so this is just checking for the `RestartContainer` policy now, which isn't changing.